src/main/src/main/java/org/geoserver/catalog/impl/CatalogImpl.java

```diff
@@ -680,10 +680,12 @@ public ValidationResult validate( LayerInfo layer, boolean isNew) {
         
         // clean up eventual dangling references to missing alternate styles
         Set<StyleInfo> styles = layer.getStyles();
-        for (Iterator it = styles.iterator(); it.hasNext();) {
-            StyleInfo styleInfo = (StyleInfo) it.next();
-            if(styleInfo == null) {
-                it.remove();
+        if (styles != null) {
```

Comment:
The styles collection is a "live" one, can never be null (there is no setter, so no way to set it to a non null value), if it happens, then a significant bug has been introduced or was already present, and should be fixed instead.